src/repository/models.py

```diff
@@ -0,0 +1,1033 @@
+    email_address = models.EmailField(unique=True)
+    first_name = models.CharField(max_length=255)
+    middle_name = models.CharField(max_length=255, blank=True, null=True)
+    last_name = models.CharField(max_length=255)
+    affiliation = models.TextField(blank=True, null=True)
+    orcid = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        verbose_name=_('ORCID')
+    )
+
+    @property
+    def full_name(self):
+        if not self.middle_name:
+            return '{} {}'.format(self.first_name, self.last_name)
+        else:
+            return '{} {} {}'.format(
+                self.first_name,
+                self.middle_name,
+                self.last_name,
+            )
+
+
+class PreprintVersion(models.Model):
+    preprint = models.ForeignKey(Preprint)
+    file = models.ForeignKey(PreprintFile)
+    version = models.IntegerField(default=1)
+    date_time = models.DateTimeField(default=timezone.now)
+    moderated_version = models.ForeignKey(
+        'VersionQueue',
+        blank=True,
+        null=True,
+    )
+    title = models.CharField(
+        max_length=300,
+        help_text=_('Your article title'),
+    )
+    abstract = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_(
+            'Please avoid pasting content from word processors as they can add '
+            'unwanted styling to the abstract. You can retype the abstract '
+            'here or copy and paste it into notepad/a plain text editor before '
+            'pasting here.',
+        )
+
+    )
+    published_doi = models.URLField(
+        max_length=255,
+        blank=True,
+        null=True,
+        verbose_name="Published Article DOI",
+        help_text="Please use the following format for your DOI: https://doi.org/10.xxxx/xxxx",
+    )
+
+    class Meta:
+        ordering = ('-version', '-date_time', '-id')
+
+    def html(self):
+        if self.file.mime_type in files.HTML_MIMETYPES:
+            return self.file.contents()
+        else:
+            return ''
+
+
+class Comment(models.Model):
+    author = models.ForeignKey(
+        'core.Account',
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+    preprint = models.ForeignKey(
+        Preprint,
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+    reply_to = models.ForeignKey(
+        'self',
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+    date_time = models.DateTimeField(default=timezone.now)
+    body = models.TextField(verbose_name='Write your comment:')
+    is_reviewed = models.BooleanField(default=False)
+    is_public = models.BooleanField(default=False)
+
+    class Meta:
+        ordering = ('-date_time', '-pk')
+
+    def __str__(self):
+        return 'Comment by {author} on {article}'.format(
+            author=self.author.full_name(),
+            article=self.preprint.title,
+        )
+    
+    def mark_public(self):
```

Comment:
`toggle_public` sounds like a better name, since it can also mark as not public this metod